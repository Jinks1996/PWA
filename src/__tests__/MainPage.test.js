import { shallow, mount, render } from "enzyme";
import MainPage from "../components/MainPage";

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 1,
        name: "John Snow",
        username: "John",
        email: "john@gmail.com",
      },
    ],
    searchField: "John",
    isPending: false,
  };

  wrapper = shallow(<MainPage {...mockProps} />);
});

// snapshot testing
it("expects to render Main Page Component", () => {
  expect(wrapper).toMatchSnapshot();
});

it("Should filter robots correctly", () => {
  expect(wrapper.instance().filteredRobots()).toEqual([
    {
      id: 1,
      name: "John Snow",
      username: "John",
      email: "john@gmail.com",
    },
  ]);
});
