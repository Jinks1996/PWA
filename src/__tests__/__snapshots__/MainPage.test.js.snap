// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render Main Page Component 1`] = `ShallowWrapper {}`;
